TaskBroker/VstsServerTaskBroker/VstsTaskHttpRetryer.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Threading.Tasks;
+using System.Web;                    
+
+namespace VstsServerTaskBroker
+{
+    /// <summary>
+    /// A class that is responsible for retrying an action if it throws an exception.
+    /// </summary>
+    public class VstsTaskHttpRetryer
```

Comment:
Why are we doing this? Our client library already has support for retries, knowing transient failures etc.